File: android/app/src/main/java/com/dibay/app/nativevideo/NativeVideoCallLog.java
package com.dibay.app.nativevideo;

import android.util.Log;

/** Native video runtime logs. Do not reuse DIBAY_CALL_V4 markers in this package. */
public final class NativeVideoCallLog {
  public static final String TAG = "DIBAY_NATIVE_VIDEO";
  private static final String PREFIX = "[DIBAY_NATIVE_VIDEO] ";

  private NativeVideoCallLog() {}

  public static void info(String marker, String callId) {
    info(marker, callId, "");
  }

  public static void info(String marker, String callId, String details) {
    Log.i(TAG, format(marker, callId, details));
    String alias = qaAlias(marker);
    if (alias != null) {
      Log.i(TAG, format(alias, callId, details));
    }
  }

  public static void warn(String marker, String callId, String details) {
    Log.w(TAG, format(marker, callId, details));
  }

  /** CUT1/CUT2/CUT4 evidence correlation — no tokens; grep DIBAY_CALL_CORR. */
  public static void corr(String marker, String callId, String details) {
    String extra =
        "marker="
            + (marker != null ? marker.trim() : "unknown")
            + " wall_ms="
            + System.currentTimeMillis()
            + (details != null && !details.trim().isEmpty() ? " " + details.trim() : "");
    Log.i(TAG, format("DIBAY_CALL_CORR", callId, extra));
  }

  private static String format(String marker, String callId, String details) {
    String sid = callId != null && !callId.trim().isEmpty() ? callId.trim() : "unknown";
    String extra = details != null && !details.trim().isEmpty() ? " " + details.trim() : "";
    return PREFIX + marker + " callId=" + sid + extra;
  }

  private static String qaAlias(String marker) {
    if ("agora_native_join_start".equals(marker)) return "agora_native_video_join_start";
    if ("agora_native_join_success".equals(marker)) return "agora_native_video_join_success";
    if ("local_camera_preview_started".equals(marker)) return "local_camera_publish_success";
    if ("remote_video_render_ready".equals(marker)) return "remote_video_rendered";
    if ("caller_agora_native_join_start".equals(marker)) return "caller_native_video_join_start";
    if ("caller_local_camera_preview_started".equals(marker)) return "caller_local_camera_publish_success";
    return null;
  }
}

File: ios/App/App/Call/Video/NativeVideoCallLog.swift
import Foundation
import os.log

/**
 * Phase B0 — Native video runtime logs.
 * Do not reuse DIBAY_CALL_V4 markers in this package.
 *
 * Device capture: same dual-write pattern as DibayCallLog
 * (os_log %{public}@ + NSLog). Console filter: subsystem `com.dibay.app`,
 * category `native_video`. Marker names / callId formatting unchanged.
 */
enum NativeVideoCallLog {
  private static let prefix = "[DIBAY_NATIVE_VIDEO] "
  private static let osLog = OSLog(subsystem: "com.dibay.app", category: "native_video")

  static func info(_ marker: String, callId: String) {
    info(marker, callId: callId, details: "")
  }

  static func info(_ marker: String, callId: String, details: String) {
    emit(level: .info, body: format(marker: marker, callId: callId, details: details))
    if let alias = qaAlias(marker) {
      emit(level: .info, body: format(marker: alias, callId: callId, details: details))
    }
  }

  static func warn(_ marker: String, callId: String, details: String) {
    emit(level: .default, body: format(marker: marker, callId: callId, details: details))
  }

  /** CUT1 evidence correlation — no tokens; grep DIBAY_CALL_CORR. */
  static func corr(_ marker: String, callId: String, details: String = "") {
    let wall = Int(Date().timeIntervalSince1970 * 1000)
    let extra =
      "marker=\(marker) wall_ms=\(wall)"
      + (details.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
        ? ""
        : " \(details.trimmingCharacters(in: .whitespacesAndNewlines))")
    emit(level: .info, body: format(marker: "DIBAY_CALL_CORR", callId: callId, details: extra))
  }

  private static func emit(level: OSLogType, body: String) {
    let line = prefix + body
    os_log("%{public}@", log: osLog, type: level, line)
    NSLog("%@", line)
  }

  private static func format(marker: String, callId: String, details: String) -> String {
    let sid = callId.trimmingCharacters(in: .whitespacesAndNewlines)
    let resolved = sid.isEmpty ? "unknown" : sid
    let extra = details.trimmingCharacters(in: .whitespacesAndNewlines)
    if extra.isEmpty {
      return "\(marker) callId=\(resolved)"
    }
    return "\(marker) callId=\(resolved) \(extra)"
  }

  private static func qaAlias(_ marker: String) -> String? {
    switch marker {
    case "agora_native_join_start":
      return "agora_native_video_join_start"
    case "agora_native_join_success":
      return "agora_native_video_join_success"
    case "local_camera_preview_started":
      return "local_camera_publish_success"
    case "remote_video_render_ready":
      return "remote_video_rendered"
    case "caller_agora_native_join_start":
      return "caller_native_video_join_start"
    case "caller_local_camera_preview_started":
      return "caller_local_camera_publish_success"
    default:
      return nil
    }
  }
}
